src/main/kotlin/uk/gov/dwp/dataworks/aws/AwsCommunicator.kt

```diff
@@ -268,9 +274,28 @@ class AwsCommunicator {
     /**
      * Deletes the [Service] from the cluster [clusterName] with the name [serviceName]. If
      * the ECS service does not exist, this function will take no action other than logging this fact.
+     *
+     * Before deleting the service, we have to wait until all tasks are stopped within it. To
+     * achieve this, we issue a scale to 0 request and poll until the service has 0 running tasks.
      */
     fun deleteEcsService(correlationId: String, clusterName: String, serviceName: String) {
         try {
+            awsClients.ecsClient.updateService(UpdateServiceRequest.builder()
+                    .cluster(clusterName)
+                    .service(serviceName)
+                    .desiredCount(0)
+                    .build()).service()
+
+            while (true) {
+                Thread.sleep(5000)
+                val service = awsClients.ecsClient.describeServices(DescribeServicesRequest.builder()
+                        .cluster(clusterName)
+                        .services(serviceName).build()).services()[0]
+                if(service.runningCount() == 0) {
+                    break
+                }
+            }
```

Comment:
I assume this is needed because the `deleteService` call doesn't succeed unless there are 0 running tasks?

Reply:
Exactly yes, we need to wait for all services to not be in a RUNNING state unfortunately

Reply:
That is... disappointing 😞